server/utils/s3.ts

```diff
@@ -93,6 +93,37 @@ export const getSignature = (policy: string) => {
   return signature;
 };
 
+export const getPresignedPost = (
+  key: string,
+  acl: string,
+  contentType = "image"
+) => {
+  const params = {
+    Bucket: process.env.AWS_S3_UPLOAD_BUCKET_NAME,
+    Conditions: [
+      // @ts-expect-error ts-migrate(2532) FIXME: Object is possibly 'undefined'.
+      ["content-length-range", 0, +process.env.AWS_S3_UPLOAD_MAX_SIZE],
+      ["starts-with", "$Content-Type", contentType],
+      ["starts-with", "$Cache-Control", ""],
+    ],
+    Fields: {
+      key,
+      acl,
+    },
+    Expires: 3600,
+  };
+
+  return new Promise<AWS.S3.PresignedPost>((resolve, reject) => {
+    s3.createPresignedPost(params, (err, data) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(data);
+      }
+    });
+  });
```

Comment:
We can use [promisify](https://nodejs.org/dist/latest-v8.x/docs/api/util.html#util_util_promisify_original) here to tidy this up a bit

Reply:
It looks great, but I don't know how to fix this error.

```
 FAIL  server/routes/api/fileOperations.****.ts
  ● Test suite failed to run

    TypeError [ERR_INVALID_ARG_TYPE]: The "original" argument must be of type function. Received undefined

      26 |   signatureVersion: "v4",
      27 | });
    > 28 | const createPresignedPost = util.promisify<
         |                                  ^
      29 |   AWS.S3.PresignedPost.Params,
      30 |   AWS.S3.PresignedPost
      31 | >(s3.createPresignedPost);

      at Object.<anonymous> (server/utils/s3.ts:28:34)
      at Object.<anonymous> (server/models/Attachment.ts:2:1)
```

Reply:
@tommoor Any suggestions? I'd rather not spend too much time on fighting TypeScript's type inference limitations.

Reply:
Sorry, wanted to check it out and take a look – it's not actually so bad, it's just that in a couple of the tests `aws-s3` lib is mocked, so the `createPresignedPost` needs mocking too. Search for `jest.mock("aws-sdk"` to see the spots.

Reply:
Thanks! I was wondering why it claims to have received `undefined`.